src/application/Application.mjs

```diff
@@ -103,11 +103,22 @@ export default class Application extends Component {
     }
 
     __updateFocus() {
+        const prevFocusedComponent = this._focusPath ? this._focusPath[this._focusPath.length - 1] : undefined;
```

Comment:
This replicates code in __updateFocusRec, but maybe it's ok because we separate the concerns a bit.

Reply:
Code duplication reduced in update just pushed, adding a helper function for this.